### DISCUSSION TITLE
Timing issue with servo

### DISCUSSION THREAD

**Discussion by IeiuniumLux** [micropython/micropython#10302] [Hardware & Peripherals] [answered]
I have a [pyboard v1.1](https://store.micropython.org/product/PYBv1.1#_) with the latest [firmware v1.19.1](https://micropython.org/download/pybv11/) and when run the code below from the REPL prompt, the servo moves properly.

`>>> from pyb import Servo`
`>>> s = Servo(1)`
`>>> s.angle(80)`
`>>> s.angle(-80)`
`>>> s.angle(0)`
`>>> s.angle(90)`

However, if I run the above one after the other in one line, then only the last s.angle(90) statement is executed.

`>>> s.angle(80); s.angle(-80); s.angle(0); s.angle(90)`

The same occurs if I try to run the first line-by-line code inside a script in the SD card. Only the last s.angle(90) statement is executed.

Is this the expected behavior?

**Comment by peterhinch** [comment, marked as answer]
You need to leave time after each operation for the servo to complete its movement.
